mantidproject/mantid
# Engg:: Fitting presenter fittingRunNoChanged refactor

Description of work.
This PR refactors the logic which is run when the run number is changed in the `Fitting` tab under the Engineering Diffraction interface.

Changes include:
- Variables have been renamed to reflect their purpose.
- Several if blocks have been condensed or removed
- Some common logic has been split out into methods
- Additional code comments added or clarified and Doxygen added where missing
- Prep for additional refactoring - these ideas can be found in ~~#17301~~ #17433 

**To test:**
Using the files to test located at  `\\olympic\Babylon5\Public\David Fairbrother\z_test`:

Open Mantid - Interfaces - Engineering Diffraction
Go to `Settings` and set the focus output directory to the folder where the files are located
Go to `Fitting` and in the input box do the following

The following should work correctly and update the display to reflect the found file:
- Attempt to locate a file using the browse button
- Enter a single run number such as `256663`
- Enter a multi run input: `256663-256664`

Under the Plot Bank drop down the available options should match the available banks - for this test data this will be `256663: 1, 2, Cropped` and `256664: 1, 2, 1, 2, 3` (This is a separate bug as it currently doesn't distinguish textured and normal banks)

The following should fail and produce a clear and descriptive error:
- Use a large range `250-500` (It is capped at 200 for the moment)
- Use a descending range `256664-256663` or `150-100` for example
- Use a run number which does not exist
- Use a multi run number which partially exist (such as `256663-256665`)

If there are any ambiguous or unclear error messages please let me know so I can tweak them to be more user friendly or specific. 

Fixes #17301 and #17299 .

Does not need to be in release notes as it should be invisible to users. 

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
